CardScan/Classes/CreditCardOcr/OcrMainLoop.swift

```diff
@@ -228,7 +230,16 @@ open class OcrMainLoop : MachineLearningLoop {
         DispatchQueue.main.async { [weak self] in
             guard let self = self else { return }
             guard !self.userDidCancel else { return }
-            delegate?.showCardDetails(number: result.number, expiry: result.expiry, name: result.name)
+            switch (result.state) {
+            case MainLoopState.initial, MainLoopState.cardOnly:
+                delegate?.showNoCard()
+            case MainLoopState.ocrIncorrect:
+                delegate?.showWrongCard(number: result.number, expiry: result.expiry, name: result.name)
+            case MainLoopState.ocrOnly, MainLoopState.ocrAndCard, MainLoopState.ocrDelayForCard:
+                delegate?.showCardDetails(number: result.number, expiry: result.expiry, name: result.name)
+            case MainLoopState.finished:
+                delegate?.complete(creditCardOcrResult: result)
```

Comment:
Does this call `complete` twice?

Reply:
good catch. I'll remove the second call.